[Brand Updates] Update prologue carousel

hichamboushaba opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: woocommerce/woomobile-private#385
<!-- Id number of the GitHub issue this PR addresses. -->
<!-- Part of: # -> use this if your PR is one of many related to one issue -->

## Description
This PR updates the prologue carousel screen to match the new design (Ntkk0ER4hHt5y9KjEGIiGk-fi-2857_3313).

## Steps to reproduce
1. Uninstall the app if it's already installed
2. Open the app.

## Testing information
- Check the look of the carousel on different devices and themes.
- Confirm the transition from the carousel to the prologue looks good.

Tested on iPhone 16 and iPad 10th gen simulators, iOS 18.1 

## Screenshots
| Light | Dark |
| --- | --- |
| ![Simulator Screenshot - iPhone 16 - 2024-12-31 at 17 19 34](https://github.com/user-attachments/assets/f385a7f8-82cd-4374-9a33-22d54893e282) | ![Simulator Screenshot - iPhone 16 - 2024-12-31 at 17 19 43](https://github.com/user-attachments/assets/c5de3ed0-f2c1-4970-8550-06f1839edeae) |
| ![Simulator Screenshot - iPhone 16 - 2024-12-31 at 17 19 36](https://github.com/user-attachments/assets/95580ac0-441a-4b2f-9d9d-6eee77d768a4) | ![Simulator Screenshot - iPhone 16 - 2024-12-31 at 17 19 45](https://github.com/user-attachments/assets/ce13f224-57bd-4e0b-85cf-f0bd9f6a8c27) |
| ![Simulator Screenshot - iPhone 16 - 2024-12-31 at 17 19 38](https://github.com/user-attachments/assets/20628449-42b6-4361-a344-ca37424e16b7) | ![Simulator Screenshot - iPhone 16 - 2024-12-31 at 17 19 47](https://github.com/user-attachments/assets/fc5b6e01-4c3a-4775-8a38-504f7f09680f) |
| ![Simulator Screenshot - iPad (10th generation) - 2024-12-31 at 17 16 07](https://github.com/user-attachments/assets/10efe106-f86d-4680-bff2-5248d714ad36) | ![Simulator Screenshot - iPad (10th generation) - 2024-12-31 at 17 16 18](https://github.com/user-attachments/assets/db347d3f-0b3c-4512-b585-51369f099692) |
| ![Simulator Screenshot - iPad (10th generation) - 2024-12-31 at 17 16 09](https://github.com/user-attachments/assets/7c8baf63-3ee5-462e-b720-e2c98836c77d) | ![Simulator Screenshot - iPad (10th generation) - 2024-12-31 at 17 16 20](https://github.com/user-attachments/assets/cfb706f2-5eb0-43a1-a46b-4e46804373c7) |
| ![Simulator Screenshot - iPad (10th generation) - 2024-12-31 at 17 16 11](https://github.com/user-attachments/assets/8d8ffd77-3c7a-413a-bd49-053c7d1b7189) | ![Simulator Screenshot - iPad (10th generation) - 2024-12-31 at 17 16 21](https://github.com/user-attachments/assets/73c9af84-e4c0-45e7-9168-8d66b6537541) |

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

hichamboushaba committed:
Rename classes for better organization

hichamboushaba committed:
Update layout to make content is centered when it fits

hichamboushaba committed:
Update UI of the main carousel container

hichamboushaba committed:
Update page control tint color

hichamboushaba committed:
Make sure the logo is loaded with correct size

Otherwise the scaling doesn't look good, this is because the vector data is not preserved when using the .wooLogoImage extension.

hichamboushaba committed:
Adjust spacings

hichamboushaba committed:
Update carousel illustrations

hichamboushaba committed:
Various UI adjustments for better alignments in different screen sizes

hichamboushaba committed:
Simplify the UIPageViewController usage

We switch to using the built-in page control instead of offering a custom one.

dangermattic commented:
<!--
  1 Error: Podfile reference(s) to a bran...
  1 Warning: This PR is larger than 300 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Podfile reference(s) to a branch:<br />
<code>WordPressAuthenticator (from `https://github.com/wordpress-mobile/WordPressAuthenticator-iOS.git`, branch `wcios/prologue-customization`)</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

hichamboushaba committed:
Use official withTintColor for tinting the woo logo

The official version offers a better support for the dynamic color when the device theme changes.

hichamboushaba commented on WooCommerce/Classes/Authentication/Prologue/LoginOnboardingPageViewController.swift:
```diff
@@ -0,0 +1,98 @@
+
+/// Displays the Login Prologue carousel, populated with `LoginOnboardingPageTypeViewController` pages.
+///
+final class LoginOnboardingPageViewController: UIPageViewController {
```
While Github marks this file as new file, it's not, it's just an issue with how Github renders the diff (an ongoing [issue](https://github.com/orgs/community/discussions/8573)), for an easier review, please [exclude](https://github.com/woocommerce/woocommerce-ios/pull/14778/files/7f18b66a7cf77267a7fd93d8610f2cea2948e5e2..974ef30e4560ec9730cc7c96d6c8f40eda7acd33) the initial commit (the rename commit).

hichamboushaba commented on WooCommerce/Classes/Authentication/Prologue/LoginOnboardingPageViewController.swift:
```diff
@@ -6,8 +6,6 @@ final class LoginOnboardingPageViewController: UIPageViewController {
 
     private let pages: [UIViewController]
 
-    private let pageControl = UIPageControl()
```
Switching to using the built-in page control of the `UIPageViewController` simplifies the code quite a bit, and also improves the UI, as we don't need to apply any hardcoded spacing to the pages themselves (I mean the removal of `stackBottomMargin` below).

hichamboushaba commented on WooCommerce/Classes/Authentication/Prologue/LoginOnboardingPageViewController.swift:
```diff
@@ -19,66 +17,38 @@ final class LoginOnboardingPageViewController: UIPageViewController {
-        pageControl.currentPageIndicatorTintColor = .gray(.shade5)
-        pageControl.pageIndicatorTintColor = .wooCommercePurple(.shade50)
+    func configurePageControllerAppearance(pageControl: UIPageControl) {
+        pageControl.currentPageIndicatorTintColor = .accent
```
For the color itself, we could have just used `UIPageControler.appearance()`, but when trying to scale it by applying a `transform` to the appearance, it crashes, so instead we retrieve the page control from the subviews and keep the same logic as before.

hichamboushaba committed:
Set the bubbles image scaling to center to align with design

itsmeichigo reviewed (commented):
I tested on both simulators and an iPhone 16 Pro and found that when switching between light and dark modes, the logo color is not updated accordingly, which makes it hard to see the logo in light mode:

<img src="https://github.com/user-attachments/assets/46d06212-9b89-4b1e-9949-5d61e1793eee" width=320 />

hichamboushaba commented:
> I tested on both simulators and an iPhone 16 Pro and found that when switching between light and dark modes, the logo color is not updated accordingly, which makes it hard to see the logo in light mode:

@itsmeichigo thanks for the review, can you retest with last changes? I had this [commit](https://github.com/woocommerce/woocommerce-ios/pull/14778/commits/0f234e6833284883576a4e4b8de8729d8ea1aad2) to fix this issue, but forgot to push my changes until today.

itsmeichigo reviewed (approved):
Thanks for the update @hichamboushaba - I tested again on iPhone 16 Pro and simulators iPhone SE and iPad Mini and confirmed that the carousel now looks good in both light and dark modes.
